sbi/utils/get_nn_models.py

```diff
@@ -345,3 +350,35 @@ def classifier_nn(
         raise ValueError(f"'model' must be one of ['linear', 'mlp', 'resnet'].")
 
     return neural_net
+
+
+def _check_1d_flow_limitations(
+    model: str, output_numel: int, parameter_or_data: str = ""
+):
+    """
+    Gives a warning or error due to limitations of flows when estimating 1D densities.
+
+    For MAFs this gives a warning if the learned density is 1D, where MAFs are
+        inherently limited to Gaussian densities. For NSFs this raises an error if the
+        learned density is 1D. This is due to implementation issues.
+
+    Args:
+        model: The used density estimator, one of maf / mdn / made / nsf.
+        output_numel: Dimensionality of the space output space of the density estimator.
+        parameter_or_data: Whether the output of the density estimator is the parameter
+            space theta (which is the case for SNPE) or the data space x (which is the
+            case for SNL). Is used to add this information to the message raised by this
+            function.
+    """
+
+    if output_numel == 1 and model == "maf":
```

Comment:
As it is, the first if will be followed by checking the second if, which is not clean. I would nest both in the 1D check and even have an else with just pass. If you want, for simplicity, you can also if 1D: warn_or_error(specific_flow), i.e. you can save yourself some ceremony by testing before calling, since this only affects 1D.